ICE in `in get_tuple_expr, at rust/ast/rust-expr.h`

<!--
File: /tmp/im/a.rs
-->

auto-reduced (treereduce-rust):
````rust
//@compile-flags: -frust-incomplete-and-experimental-compiler-do-not-use
fn distance_squared() -> f32 {
    #[cfg(cfail1)]
    OKAY_RANGE.0
        + this.y;

    return;
}

````

original:
````rust
pub struct Point {
    z: i8,
    pub y: f32,
}

fn distance_squared(this: &Point) -> f32 {
    #[cfg(cfail1)]
     OKAY_RANGE.0 + this.y;

    #[add_alternative_to_arm(cfail2)]
    return CString::new(s).unwrap * this.x + this.y * this.y;
}

impl Point {
    pub fn distance_from_origin(&self) -> f32 {
        x::sum(&x)
    }
}

impl Point {
    pub fn translate(&mut self, x: isize, cfail2: f32) {
        -val
    }
}

````

Version information:

5c3e2adc695b04a199d37cacd8c23b14fdab914f

Possibly related line of code:
https://github.com/Rust-GCC/gccrs/blob/5c3e2adc695b04a199d37cacd8c23b14fdab914f/gcc/rust/ast/rust-expr.h#L1636-L1648

Command:
`/home/matthias/vcs/github/gccrs/gccrs-build/gcc/crab1 -frust-incomplete-and-experimental-compiler-do-not-use`

<details><summary><strong>Program output</strong></summary>
<p>

```
/tmp/icemaker_global_tempdir.RzqgB6tKwBKU/rustc_testrunner_tmpdir_reporting.rHOrftVDQdkA/mvce.rs:3:5: error: cannot strip expression in this position - outer attributes are never allowed before binary op exprs
    3 |     OKAY_RANGE.0
      |     ^~~~~~~~~~
/tmp/icemaker_global_tempdir.RzqgB6tKwBKU/rustc_testrunner_tmpdir_reporting.rHOrftVDQdkA/mvce.rs:3:5: error: cannot strip expression in this position - outer attributes are never allowed before binary op exprs
crab1: internal compiler error: in get_tuple_expr, at rust/ast/rust-expr.h:1642
0x2fec938 internal_error(char const*, ...)
	../../gcc/diagnostic-global-context.cc:517
0xc0dfbb fancy_abort(char const*, int, char const*)
	../../gcc/diagnostic.cc:1803
0x8fbc31 Rust::AST::TupleIndexExpr::get_tuple_expr()
	../../gcc/rust/ast/rust-expr.h:1642
0x8fbc31 Rust::AST::DefaultASTVisitor::visit(Rust::AST::TupleIndexExpr&)
	../../gcc/rust/ast/rust-ast-visitor.cc:358
0x8fbc31 Rust::AST::DefaultASTVisitor::visit(Rust::AST::TupleIndexExpr&)
	../../gcc/rust/ast/rust-ast-visitor.cc:355
0xd6ed90 void Rust::AST::DefaultASTVisitor::visit<Rust::AST::Expr>(Rust::AST::Expr&)
	../../gcc/rust/ast/rust-ast-visitor.h:412
0xd6ed90 Rust::AST::DefaultASTVisitor::visit(Rust::AST::ArithmeticOrLogicalExpr&)
	../../gcc/rust/ast/rust-ast-visitor.cc:263
0xd6e78f void Rust::AST::DefaultASTVisitor::visit<Rust::AST::Stmt>(std::unique_ptr<Rust::AST::Stmt, std::default_delete<Rust::AST::Stmt> >&)
	../../gcc/rust/ast/rust-ast-visitor.h:416
0xd6e78f Rust::AST::DefaultASTVisitor::visit(Rust::AST::BlockExpr&)
	../../gcc/rust/ast/rust-ast-visitor.cc:453
0xf7ffa8 std::function<void ()>::operator()() const
	/usr/include/c++/15.1.1/bits/std_function.h:593
0xf7ffa8 Rust::Resolver2_0::NameResolutionContext::scoped(Rust::Resolver2_0::Rib::Kind, unsigned int, std::function<void ()>, tl::optional<Rust::Identifier>)
	../../gcc/rust/resolve/rust-name-resolution-context.cc:283
0xf8471c Rust::Resolver2_0::DefaultResolver::visit(Rust::AST::BlockExpr&)
	../../gcc/rust/resolve/rust-default-resolver.cc:50
0xf88ca9 operator()
	../../gcc/rust/resolve/rust-default-resolver.cc:72
0xf88ca9 scope_inner<Rust::Resolver2_0::CanonicalPathCtx::scope<Rust::Resolver2_0::DefaultResolver::visit(Rust::AST::Function&)::<lambda()> >(Rust::NodeId, std::string, Rust::Resolver2_0::DefaultResolver::visit(Rust::AST::Function&)::<lambda()>&&)::<lambda()>, Rust::Resolver2_0::DefaultResolver::visit(Rust::AST::Function&)::<lambda()> >
	../../gcc/rust/resolve/rust-name-resolution-context.h:470
0xf88ca9 scope<Rust::Resolver2_0::DefaultResolver::visit(Rust::AST::Function&)::<lambda()> >
	../../gcc/rust/resolve/rust-name-resolution-context.h:413
0xf88ca9 scope<Rust::Resolver2_0::DefaultResolver::visit(Rust::AST::Function&)::<lambda()> >
	../../gcc/rust/resolve/rust-name-resolution-context.h:406
0xf88ca9 operator()
	../../gcc/rust/resolve/rust-default-resolver.cc:75
0xf88ca9 __invoke_impl<void, Rust::Resolver2_0::DefaultResolver::visit(Rust::AST::Function&)::<lambda()>&>
	/usr/include/c++/15.1.1/bits/invoke.h:63
0xf88ca9 __invoke_r<void, Rust::Resolver2_0::DefaultResolver::visit(Rust::AST::Function&)::<lambda()>&>
	/usr/include/c++/15.1.1/bits/invoke.h:113
0xf88ca9 _M_invoke
	/usr/include/c++/15.1.1/bits/std_function.h:292
Please submit a full bug report, with preprocessed source (by using -freport-bug).
Please include the complete backtrace with any bug report.
See <https://gcc.gnu.org/bugs/> for instructions.

```

</p>
</details>

--- a/gcc/rust/ast/rust-expr.h
+++ b/gcc/rust/ast/rust-expr.h
@@ -1611,6 +1611,7 @@ class TupleIndexExpr : public ExprWithoutBlock
   TupleIndex tuple_index;
 
   location_t locus;
+  bool to_strip;
 
   // i.e. pair.0
 
@@ -1622,13 +1623,15 @@ public:
   TupleIndexExpr (std::unique_ptr<Expr> tuple_expr, TupleIndex index,
 		  std::vector<Attribute> outer_attribs, location_t locus)
     : outer_attrs (std::move (outer_attribs)),
-      tuple_expr (std::move (tuple_expr)), tuple_index (index), locus (locus)
+      tuple_expr (std::move (tuple_expr)), tuple_index (index), locus (locus),
+      to_strip (false)
   {}
 
   // Copy constructor requires a clone for tuple_expr
   TupleIndexExpr (TupleIndexExpr const &other)
     : ExprWithoutBlock (other), outer_attrs (other.outer_attrs),
-      tuple_index (other.tuple_index), locus (other.locus)
+      tuple_index (other.tuple_index), locus (other.locus),
+      to_strip (other.to_strip)
   {
     // guard to prevent null dereference (only required if error state)
     if (other.tuple_expr != nullptr)
@@ -1642,6 +1645,7 @@ public:
     tuple_index = other.tuple_index;
     locus = other.locus;
     outer_attrs = other.outer_attrs;
+    to_strip = other.to_strip;
 
     // guard to prevent null dereference (only required if error state)
     if (other.tuple_expr != nullptr)
@@ -1661,8 +1665,8 @@ public:
   void accept_vis (ASTVisitor &vis) override;
 
   // Invalid if tuple expr is null, so base stripping on that.
-  void mark_for_strip () override { tuple_expr = nullptr; }
-  bool is_marked_for_strip () const override { return tuple_expr == nullptr; }
+  void mark_for_strip () override { to_strip = true; }
+  bool is_marked_for_strip () const override { return to_strip; }
 
   // TODO: is this better? Or is a "vis_block" better?
   Expr &get_tuple_expr ()

